clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -20449,23 +20449,44 @@ RValue CodeGenFunction::EmitIntelFPGAMemBuiltin(const CallExpr *E) {
   // Arguments
   const Expr *PtrArg = E->getArg(0);
   Value *PtrVal = EmitScalarExpr(PtrArg);
+  ASTContext &Ctx = getContext();
 
   // Create the pointer annotation
   Function *F =
       CGM.getIntrinsic(llvm::Intrinsic::ptr_annotation, PtrVal->getType());
   SmallString<256> AnnotStr;
   llvm::raw_svector_ostream Out(AnnotStr);
 
-  Optional<llvm::APSInt> Params =
-      E->getArg(1)->getIntegerConstantExpr(getContext());
+  Optional<llvm::APSInt> Params = E->getArg(1)->getIntegerConstantExpr(Ctx);
```

Comment:
May we use AddOptionalArgValue for first two parameters as well (maybe to rename lambda to AddArgValue)?